### ISSUE TITLE
Prevent multiple statements on the same line

### ISSUE THREAD

**Issue by hiqua** [pinterest/ktlint#1078]
## Duplicates
I could not find a matching issue, but I find it hard to believe that it doesn't exist yet. Apologies if this is indeed a duplicate.

## Expected Rule behavior
Do not allow multiple statements on the same line. ktlint does not complain with the following:
```kotlin
class MissingRules {
    fun f() {
        val a = 3; val b = 2
    }
}
```
Here, `a` and `b` should be declared separately.

## Additional information
* Current version of ktlint: 0.40.0
* Styleguide section: https://developer.android.com/kotlin/style-guide#one_statement_per_line
```
One statement per line
Each statement is followed by a line break. Semicolons are not used.
```

**Comment by paul-dingemans** [comment]
I am not aware of any language constructs in Kotlin that use `;` as separator between elements.

**Comment by atulgpt** [comment]
Hi, @paul-dingemans Should we disallow the semicolon itself? Even code like the below is possible in Kotlin. Which are not reported by `ktlint`
```
public fun f1() {
    val a = 3; val b = 2
}; public fun f2() {
    val a = 3; val b = 2
}
```
```
f1(); f2()
```
```
public class A {
    
}; public class B {
    
}
```
```
public class A; public class B
```
